### ISSUE TITLE
Tensorflow can run and build model with the corner case `Dense(unit=0)`

### ISSUE THREAD

**Issue by shiningrain** [tensorflow/tensorflow#37335]
**System information**  
- Have I written custom code (as opposed to using example directory):  
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):  Windows 10
- TensorFlow backend (yes / no):  yes
- TensorFlow version:  1.15.0
- Python version:  3.6.9
- CUDA/cuDNN version:  -
- GPU model and memory:  -

**Describe the current behavior**  
When I set `Dense (units = 0)`, **TensorFlow can build the model normally**. `units = 0` is an obviously  unreasonable parameter, which should be stopped before building model. But Tensorflow treats it as a normal parameter. The model saved by Tensroflow may lead to potential risk. 
Does unit = 0 have any special effect on TensorFlow? I don't see the corresponding instructions in the documentation. If not,**should Tensorflow set a check for such unreasonable parameters to avoid the risks and incorrect usages in the model?** 

**Code to reproduce the issue**  
```
import numpy as np
import keras.layers as L
import keras.backend as K
from keras.engine import Model, Input

## Using Tensorflow as Keras backend.
## Input dtype default is float32

kwargs={'units': 0}
input = (10 * np.random.random((1,32,32,16)))
layer = L.core.Dense(**kwargs)
x = Input(batch_shape=input.shape)
y = layer(x)
bk_model = Model(x, y)
print('finish')
```

**Comment by Saduf2019** [comment]
@shiningrain 
There is a duplicate ticket with same code/issue #37336, could you please confirm if we may close this issue as duplicate as it is been monitored there.

**Comment by shiningrain** [comment]
> @shiningrain
> There is a duplicate ticket with same code/issue #37336, could you please confirm if we may close this issue as duplicate as it is been monitored there.

 Hello, thank you very much for your reply. 
 Both of this question and [issue 37336](https://github.com/tensorflow/tensorflow/issues/37336)   hope that the problem can be reproduced with the shortest possible code, thus the structure of the codes are similar. **But they reflect totally different problems**.  This issue focuses on **‘Tensorflow may lack a detection of improper parameters when establishing `Dense`’**, while the point of [issue 37336](https://github.com/tensorflow/tensorflow/issues/37336) is **‘Error reported when building GRU model under certain parameters’**.
  These are two different problems, and the code structure is also designed to **make the problem easier to be analyzed and reproduced**.
     Hope you can understand me, thank you again for your reply

**Comment by ymodak** [comment]
Can you please change your imports to `tensorflow.keras` rather than `keras`?
```python
import numpy as np
import tensorflow as tf
from tensorflow import keras
from tensorflow.keras import layers, Model

## Using Tensorflow as Keras backend.
## Input dtype default is float32

kwargs={'units': 0}
input = (10 * np.random.random((1,32,32,16)))
layer = layers.Dense(**kwargs)
x = keras.Input(batch_shape=input.shape)
y = layer(x)
bk_model = Model(x, y)
print('finish')
```

**Comment by shiningrain** [comment]
> ```python
> import numpy as np
> import tensorflow as tf
> from tensorflow import keras
> from tensorflow.keras import layers, Model
> 
> ## Using Tensorflow as Keras backend.
> ## Input dtype default is float32
> 
> kwargs={'units': 0}
> input = (10 * np.random.random((1,32,32,16)))
> layer = layers.Dense(**kwargs)
> x = keras.Input(batch_shape=input.shape)
> y = layer(x)
> bk_model = Model(x, y)
> print('finish')
> ```

Thanks for your help!
I have tried your code, and **it can still build and save this dangerous model**, shown in the following picture:
![image](https://user-images.githubusercontent.com/46860123/76579839-a88a4880-6508-11ea-9db0-be4e44b22f7d.png)
Thank you again!

**Comment by ymodak** [comment]
This is fixed with TF 2.4. Thanks!
See [gist](https://colab.research.google.com/gist/ymodak/b15f37fe712fa2825c96012b8b96f67b/untitled637.ipynb)

**Comment by tensorflowbutler** [comment]
Hi There,

 We are checking to see if you still need help on this, as you are using an older version of tensorflow which is officially considered end of life . We recommend that you upgrade to the latest 2.x version and let us know if the issue still persists in newer versions. Please open a new issue for any help you need against 2.x, and we will get you the right help. 

 This issue will be closed automatically 7 days from now. If you still need help with this issue, please provide us with more information.
